File: src/main/java/buttondevteam/discordplugin/commands/ICommand2DC.java
package buttondevteam.discordplugin.commands;

import buttondevteam.discordplugin.DiscordPlugin;
import buttondevteam.lib.chat.CommandClass;
import buttondevteam.lib.chat.ICommand2;
import lombok.Getter;
import lombok.val;

public abstract class ICommand2DC extends ICommand2 {
	public <T extends ICommand2> ICommand2DC() {
		super(DiscordPlugin.plugin.getManager());
		val ann = getClass().getAnnotation(CommandClass.class);
		if (ann == null)
			modOnly = false;
		else
			modOnly = ann.modOnly();
	}

	private final @Getter boolean modOnly;
}

File: src/main/java/buttondevteam/discordplugin/commands/ICommand2DC.scala
package buttondevteam.discordplugin.commands

import buttondevteam.discordplugin.DiscordPlugin
import buttondevteam.lib.chat.{CommandClass, ICommand2}

abstract class ICommand2DC() extends ICommand2[Command2DCSender](DiscordPlugin.plugin.manager) {
    final private var modOnly = false

    {
        val ann: CommandClass = getClass.getAnnotation(classOf[CommandClass])
        if (ann == null) modOnly = false
        else modOnly = ann.modOnly
    }

    def isModOnly: Boolean = this.modOnly
}